src/Symfony/Component/Translation/DependencyInjection/TranslatorPass.php

```diff
@@ -55,8 +55,13 @@ public function process(ContainerBuilder $container)
 
             foreach ($container->findTaggedServiceIds('validator.constraint_validator', true) as $id => $attributes) {
                 $serviceDefinition = $container->getDefinition($id);
+                $className = $serviceDefinition->getClass();
+                // Resolve Constraint Validator FQCN defined as %foo.bar.class% parameter
+                if (preg_match('/^%([^%]+)%$/', $className, $match)) {
+                    $className = $container->getParameter($match[1]);
+                }
```

Comment:
There's `$container->getParameterBag()->resolveValue()` so you don't have to resolve parameters by yourself.